test_mq_service.py

```diff
@@ -227,26 +233,20 @@ def test_handle_message_verify_metadata_pass_through(self):
             expected_results["platform_metadata"],
         )
 
-    def test_handle_message_verify_metadata_is_not_required(self):
-        expected_insights_id = str(uuid.uuid4())
-
-        host_data = {
-            "display_name": "test_host",
-            "insights_id": expected_insights_id,
-            "account": "0000001",
-            "stale_timestamp": "2019-12-16T10:10:06.754201+00:00",
-            "reporter": "test",
-        }
+    @patch("app.queue.ingress.host_repository.add_host")
+    def test_handle_message_verify_message_key_and_metadata_not_required(self, add_host):
+        add_host.return_value = (self._host_data(), AddHostResults.created)
 
-        message = {"operation": "add_host", "data": host_data}
+        message = {"operation": "add_host", "data": self._host_data()}
 
-        expected_results = {"host": {**host_data}}
+        expected_results = {"host": {**add_host.return_value[0]}}
 
         mock_event_producer = mockEventProducer()
         with self.app.app_context():
             handle_message(json.dumps(message), mock_event_producer)
+            self.assertEqual(mock_event_producer.get_message_key(), expected_results["host"]["id"])
 
-        for key in host_data.keys():
+        for key in expected_results["host"].keys():
             self.assertEqual(
                 json.loads(mock_event_producer.get_write_event())["host"][key], expected_results["host"][key]
             )
```

Comment:
No no no.

- You are unmarshalling the whole event JSON string, which is an expensive operation, for every host key. It is possible to do the _json.loads_ just once before the loop.
- The loop itself is unnecessary. We can compare the whole `event["host"]` and `host_data` dictionaries directly.

```suggestion
        event = json.loads(mock_event_producer.event)
        self.assertEqual(event["host"], host_data)
```